src/phase1.hpp

```diff
@@ -266,20 +266,23 @@ void* phase1_thread(THREADDATA* ptd)
                 // so now we can compare entries in both buckets to find matches. If two entries
                 // match, match, the result is written to the right table. However the writing
                 // happens in the next iteration of the loop, since we need to remap positions.
-                std::vector<std::pair<uint16_t, uint16_t>> match_indexes;
+                // std::vector<std::pair<uint16_t, uint16_t>> match_indexes;
+                uint16_t idx_L[10000];
+                uint16_t idx_R[10000];
```

Comment:
I would also be a bit worried about the fixed size of this array. There isn't a way to reliably compute an upper bound of this? At the very least, I think `idx_count` should be checked after the call to `FindMatches()` to ensure its < `std::size(idx_R)` and abort if it isn't. We would have invoked undefined behavior already, but at least we would have a chance to abort

Reply:
checked